[BUG] 'ScheduleLambda' Rejects Closures Capturing 64-bit-aligned Values on 32-bit (ILP32) targets

### Reproduction steps

### Problem

`CHIP_CONFIG_LAMBDA_EVENT_ALIGN` (_src/lib/core/CHIPConfig.h_) defaults to `sizeof(void *)`. This is a pointer *size* being used as the *alignment* of the `System::Layer` lambda event storage. On LP64 the two happen to coincide (8), covering all common scalar captures, so the default is fine. On ILP32 it is 4, which is smaller than `alignof(int64_t) == 8`.

Any lambda posted via `System::Layer::ScheduleLambda` that captures a 64-bit-aligned value therefore fails to compile, via the assertion in `src/lib/support/LambdaBridge.h`:

```
    static_assert(CHIP_CONFIG_LAMBDA_EVENT_ALIGN % alignof(Lambda) == 0,
                  "lambda align too large");
```

with `4 % 8 != 0`.

### Reproduction steps

On any 32-bit target (for example, arm-none-linux-gnueabihf):

```
    uint64_t value = ...;
    chip::DeviceLayer::SystemLayer().ScheduleLambda(
        [value]() { /* ... */ });
```

Compilation fails:

```
    LambdaBridge.h:36: error: static assertion failed: lambda align too large
    note: the comparison reduces to '(4 == 0)'
```

### Expected

`ScheduleLambda` accepts closures capturing fundamental-aligned types up to the platform's maximum alignment. Capturing a 64-bit value (timestamp, counter, energy/volume measurement) is common and should not require per-target reconfiguration.

### Actual

Compile-time failure on ILP32 whenever the closure's alignment exceeds `sizeof(void *)`.

### Root cause

`sizeof(void *)` is a size standing in for an alignment; it is only incidentally correct on LP64. The correct basis is an alignment that covers every fundamental scalar, this is `alignof(std::max_align_t)`.

### Environment

- Target: 32-bit ARM (arm-none-linux-gnueabihf), but applies to any ILP32 ABI
- Toolchain: GCC 12.3.1
- connectedhomeip: abcc720b48c5e59c0edcfe65c516f76ca9448aa3

### Proposed fix

Define `CHIP_CONFIG_LAMBDA_EVENT_ALIGN` as `alignof(std::max_align_t)` (adding `<cstddef>`).

PR to follow.

### Bug prevalence

Always

### GitHub hash of the SDK that was being used

abcc720b48c5e59c0edcfe65c516f76ca9448aa3

### Platform

linux

### Platform Version(s)

1.5.1.0

### Anything else?

_No response_

--- a/src/lib/core/CHIPConfig.h
+++ b/src/lib/core/CHIPConfig.h
@@ -39,6 +39,8 @@
 #include <core/CHIPBuildConfig.h>
 #endif
 
+#include <cstddef>
+
 #include <ble/BleConfig.h>
 #include <system/SystemConfig.h>
 
@@ -1049,9 +1051,19 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
  * @def CHIP_CONFIG_LAMBDA_EVENT_ALIGN
  *
  * @brief The maximum alignment of the lambda which can be post into system event queue.
+ *
+ * @note For those platforms in which memory is not constrained,
+ * express the default as an alignment rather than a size, using
+ * alignof. The alignment that, by definition, covers every
+ * fundamental scalar a closure can capture, including `long
+ * double`. Otherwise, default to pointer alignment.
  */
 #ifndef CHIP_CONFIG_LAMBDA_EVENT_ALIGN
-#define CHIP_CONFIG_LAMBDA_EVENT_ALIGN (sizeof(void *))
+#if (defined(__linux__) && __linux__) || (defined(__MACH__) && __MACH__)
+#define CHIP_CONFIG_LAMBDA_EVENT_ALIGN (alignof(std::max_align_t))
+#else
+#define CHIP_CONFIG_LAMBDA_EVENT_ALIGN (alignof(void *))
+#endif
 #endif
 
 /**

